src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -149,14 +152,27 @@ public void deleteCollection(String collectionName) throws PineconeException {
         }
     }
 
-    public Index createIndexConnection(String indexName) {
-        PineconeConnection connection = new PineconeConnection(config, indexName);
-        return new Index(connection);
+    public Index getIndexConnection(String indexName) {
+        config.setHost(getIndexHost(indexName));
+        PineconeConnection connection = getConnection(indexName);
+        return new Index(this, connection, indexName);
+    }
+
+    public AsyncIndex getAsyncIndexConnection(String indexName) {
+        PineconeConnection connection = getConnection(indexName);
+        return new AsyncIndex(this, connection, indexName);
+    }
+
+    private PineconeConnection getConnection(String indexName) {
+        return connectionsMap.computeIfAbsent(indexName, key -> new PineconeConnection(config));
+    }
+
+    ConcurrentHashMap<String, PineconeConnection> getConnectionsMap() {
+        return connectionsMap;
     }
```

Comment:
Is this primarily for testing, or do we expect users to interact with this?

Reply:
Yes, for testing only